packages/kit/src/utils/url.js

```diff
@@ -54,23 +54,20 @@ export function normalize_path(path, trailing_slash) {
 	return path;
 }
 
+/**
+ * Decode pathname excluding %25 to prevent further double decoding of params
+ * @param {string} pathname
+ */
+export function decode_pathname(pathname) {
+	return pathname.split('%25').map(decodeURI).join('%25');
```

Comment:
should this be `decodeURIComponent`?

```js
decodeURI('%40'); // '%40'
decodeURIComponent('%40'); // '@'
```

```suggestion
	return pathname.split('%25').map(decodeURIComponent).join('%25');
```

Reply:
no, `decodeURIComponent` will also do `%2f` -> `/`, which will break further candidate patterns regex here:

https://github.com/sveltejs/kit/blob/a6a8690edf6ae21bdba1c09aa2e5a77e740ebec8/packages/kit/src/runtime/server/index.js#L72

Reply:
ah, gotcha